File: examples-sdk/csharp/607-db-vilorm-multitenant/Main.cs
// 607-db-vilorm-multitenant — C# SDK equivalent
// Compile: vil compile --from csharp --input 607-db-vilorm-multitenant/Main.cs --release

#r "sdk/csharp/Vil.cs"

var server = new VilServer("vilorm-multitenant", 8087);
var saas = new ServiceProcess("saas");
saas.Endpoint("POST", "/tenants", "create_tenant");
saas.Endpoint("GET", "/tenants/:id", "get_tenant");
saas.Endpoint("PUT", "/tenants/:id", "update_tenant");
saas.Endpoint("POST", "/tenants/:id/users", "add_user");
saas.Endpoint("GET", "/tenants/:id/users", "list_users");
saas.Endpoint("POST", "/tenants/:id/settings", "upsert_setting");
saas.Endpoint("GET", "/tenants/:id/settings", "list_settings");
saas.Endpoint("GET", "/tenants/:id/stats", "tenant_stats");
server.Service(saas);
server.Compile();

File: examples-sdk/swift/607-db-vilorm-multitenant/main.swift
// 607-db-vilorm-multitenant — Swift SDK equivalent
// Compile: vil compile --from swift --input 607-db-vilorm-multitenant/main.swift --release

let server = VilServer(name: "vilorm-multitenant", port: 8087)
let saas = ServiceProcess(name: "saas")
saas.endpoint(method: "POST", path: "/tenants", handler: "create_tenant")
saas.endpoint(method: "GET", path: "/tenants/:id", handler: "get_tenant")
saas.endpoint(method: "PUT", path: "/tenants/:id", handler: "update_tenant")
saas.endpoint(method: "POST", path: "/tenants/:id/users", handler: "add_user")
saas.endpoint(method: "GET", path: "/tenants/:id/users", handler: "list_users")
saas.endpoint(method: "POST", path: "/tenants/:id/settings", handler: "upsert_setting")
saas.endpoint(method: "GET", path: "/tenants/:id/settings", handler: "list_settings")
saas.endpoint(method: "GET", path: "/tenants/:id/stats", handler: "tenant_stats")
server.service(saas)
server.compile()
